### PR TITLE
Fix client no retry on good token

### PR DESCRIPTION
**Pull request by cviecco**
1. The okta 2fa cope did not return an error message when no valid autneticators where available
2. When both mfa AND token failed the client did not termiate on error conditins

### CODE REVIEW

File: cmd/keymaster/main.go [hunk 1]
```diff
@@ -362,6 +362,7 @@ func setupCerts(
 
 		}
 	}
+	logger.Debugf(1, "SetupCerts: authenticaiton Complete")
```
Comment (rgooch) [inline cmd/keymaster/main.go:365]: s/authenticaiton/authentication/

File: cmd/keymasterd/2fa_okta.go [hunk 2]
```diff
@@ -95,12 +95,24 @@ func (state *RuntimeState) oktaPushStartHandler(w http.ResponseWriter, r *http.R
 		state.writeFailureResponse(w, r, http.StatusInternalServerError, "Apperent Misconfiguration")
 		return
 	}
+	userResponse, err := oktaAuth.GetValidUserResponse(authData.Username)
+	if err != nil {
+		logger.Debugf(2, "oktaPushStartHandler: ")
+	}
+	if len(userResponse.Embedded.Factor) < 1 {
+		logger.Printf("oktaPushStartHandler: user %s does not have valid authenticators", authData.Username)
+		logger.Debugf(2, "oktaPushStartHandler: usedata for broken user%s is :%s", authData.Username, userResponse)
```
Comment (rgooch) [inline cmd/keymasterd/2fa_okta.go:104]: s/usedata/userdata/ ?

File: lib/authenticators/okta/impl.go [hunk 3]
```diff
@@ -195,13 +195,14 @@ func (pa *PasswordAuthenticator) validateUserOTP(username string, otpValue int)
...
 func (pa *PasswordAuthenticator) validateUserPush(username string) (PushResponse, error) {
-	userResponse, err := pa.getValidUserResponse(username)
+	userResponse, err := pa.GetValidUserResponse(username)
 	if err != nil {
 		return PushResponseRejected, err
 	}
 	if userResponse == nil {
 		return PushResponseRejected, nil
 	}
+	pa.logger.Debugf(2, "oktaAuthenticator: validsteUserPush: after getting userResponse=%+v", userResponse)
```
Comment (rgooch) [inline lib/authenticators/okta/impl.go:205]: `validste`?

**Review by rgooch — CHANGES_REQUESTED** [review]

**Comment by rgooch** [comment]
Is this PR still needed? Review comments are still open.

**Comment by cviecco** [comment]
Closed with new revision reopened

**Comment by rgooch** [comment]
Changes in PR #240.

**Pull request closed without merging**